### PR TITLE
CI-6595:Added condition to run more than 1 GHA in parallel

### PR DESCRIPTION
**Pull request by sahithibanda01**
image

### CODE REVIEW

File: docs/continuous-integration/use-ci/use-drone-plugins/ci-github-action-step.md [hunk 1]
```diff
@@ -151,6 +151,7 @@ For [private Action repositories](#private-action-repositories), you must provid
 * If you already configured GitHub Actions elsewhere, you can quickly [transfer GitHub Actions into Harness CI](#transfer-github-actions-into-harness-ci) by copying the `spec` details from your existing GitHub Actions YAML.
 * You can use variable expressions in the `with` and `env` settings. For example, `credentials: <+stage.variables.[TOKEN_SECRET]>` uses a [stage variable](/docs/platform/Pipelines/add-a-stage#option-stage-variables).
 * For GitHub Actions steps, `with` mappings are automatically exported as [output variables](#output-variables-from-github-actions-steps).
+* If there are multiple GitHub Actions with same action name and is being used in parallel. Nektos not able to clone correctly , it has some race condition due to which failure happens But if we change the directories(by specifying environment variables(For example: (XDG_CACHE_HOME,  /home/ubuntu/.cache1)) ) we can be able to run pipeline.
```
Comment (raghavharness) [inline docs/continuous-integration/use-ci/use-drone-plugins/ci-github-action-step.md:154]: I would let's not explain the internal problem that much to the user. We can just mention something like
`
Please set XDG_CACHE_HOME environment variable if you are running the same github action parallelly or by using a looping strategy. The variable XDG_CACHE_HOME should have different value for each step for the execution to work seamlessly. The default value of this variable is /home/ubuntu/.cache. You can set any different value for example /home/ubuntu/.cache1.
`

File: docs/continuous-integration/use-ci/use-drone-plugins/ci-github-action-step.md [hunk 2]
```diff
@@ -142,14 +142,17 @@ The `spec` parameters define which Action to use, the Action settings, and envir
...
+
+* **Private Action repos:** For [private Action repositories](#private-action-repositories), you must provide the `GITHUB_TOKEN` environment variable, such as `GITHUB_TOKEN: <+secrets.getValue("[SECRET_NAME]")>`. You need a [GitHub personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) that has pull permissions to the target repository. Additional permissions may be necessary depending on the Action's purpose.
+* **Duplicate Actions:** If you run multiple instances of the same GitHub Action, either in parallel or with a looping strategy, you must set the `XDG_CACHE_HOME` environment variable. The default value of this variable is `/home/ubuntu/.cache`; however, the `XDG_CACHE_HOME` variable must have a different value for each iteration of the Action. If you have separate steps running in parallel, assign distinct values to each step, such as `XDG_CACHE_HOME: /home/ubuntu/.cache1`. If you apply a looping strategy to the same step, you can use an expressions to generate distinct values, such as `XDG_CACHE_HOME: /home/ubuntu/.cache<+step.identifier>`.
```
Comment (brian-f-harness) [inline docs/continuous-integration/use-ci/use-drone-plugins/ci-github-action-step.md:150]:
```suggestion
* **Duplicate Actions:** If you run multiple instances of the same GitHub Action, either in parallel or with a looping strategy, you must set the `XDG_CACHE_HOME` environment variable. The default value of this variable is `/home/ubuntu/.cache`; however, the `XDG_CACHE_HOME` variable must have a different value for each iteration of the Action. If you have separate steps running in parallel, assign distinct values to each step, such as `XDG_CACHE_HOME: /home/ubuntu/.cache1`. If you apply a looping strategy to the same step, you can use an expression to generate distinct values, such as `XDG_CACHE_HOME: /home/ubuntu/.cache<+step.identifier>`.
```

File: docs/continuous-integration/use-ci/use-drone-plugins/ci-github-action-step.md [hunk 3]
```diff
@@ -174,20 +177,22 @@ For [private Action repositories](#private-action-repositories), you must provid
...
 
+   The following cases *always* require environment variables:
+
+   * **Private Action repos:** For [private Action repositories](#private-action-repositories), you must provide the `GITHUB_TOKEN` environment variable. You need a [GitHub personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) that has pull permissions to the target repository. Additional permissions may be necessary depending on the Action's purpose.
+   * **Duplicate Actions:** If you run multiple instances of the same GitHub Action, either in parallel or with a looping strategy, you must set the `XDG_CACHE_HOME` environment variable. The default value of this variable is `/home/ubuntu/.cache`; however, the `XDG_CACHE_HOME` variable must have a different value for each iteration of the Action. If you have separate steps running in parallel, assign distinct values to each step, such as `XDG_CACHE_HOME: /home/ubuntu/.cache1`. If you apply a looping strategy to the same step, you can use an expressions to generate distinct values, such as `XDG_CACHE_HOME: /home/ubuntu/.cache<+step.identifier>`.
```
Comment (brian-f-harness) [inline docs/continuous-integration/use-ci/use-drone-plugins/ci-github-action-step.md:185]:
```suggestion
   * **Duplicate Actions:** If you run multiple instances of the same GitHub Action, either in parallel or with a looping strategy, you must set the `XDG_CACHE_HOME` environment variable. The default value of this variable is `/home/ubuntu/.cache`; however, the `XDG_CACHE_HOME` variable must have a different value for each iteration of the Action. If you have separate steps running in parallel, assign distinct values to each step, such as `XDG_CACHE_HOME: /home/ubuntu/.cache1`. If you apply a looping strategy to the same step, you can use an expression to generate distinct values, such as `XDG_CACHE_HOME: /home/ubuntu/.cache<+step.identifier>`.
```

File: docs/continuous-integration/use-ci/use-drone-plugins/ci-github-action-step.md [hunk 4]
```diff
@@ -174,20 +177,22 @@ For [private Action repositories](#private-action-repositories), you must provid
...
-
    Refer to the GitHub Action's `env` usage specifications for details about specific settings available for the Action that you want to use. Note that `env` specifies incoming environment variables, which are separate from outgoing environment variables that may be output by the Action.
 
+   The following cases *always* require environment variables:
+
+   * **Private Action repos:** For [private Action repositories](#private-action-repositories), you must provide the `GITHUB_TOKEN` environment variable. You need a [GitHub personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) that has pull permissions to the target repository. Additional permissions may be necessary depending on the Action's purpose.
```
Comment (brian-f-harness) [inline docs/continuous-integration/use-ci/use-drone-plugins/ci-github-action-step.md:184]:
```suggestion
   * **Private Action repos:** For [private Action repositories](#private-action-repositories), you must provide the `GITHUB_TOKEN` environment variable. You need a [GitHub personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) that has pull permissions to the target repository. Additional permissions may be required, depending on the Action's purpose.
```

File: docs/continuous-integration/use-ci/use-drone-plugins/run-a-git-hub-action-in-cie.md [hunk 5]
```diff
@@ -145,6 +141,50 @@ Here's an example of the YAML for a `Plugin` step using an Action in a private r
                        GITHUB_TOKEN: <+secrets.getValue("github_pat")>
 ```
 
+### Duplicate Actions
+
+If you run multiple instances of the same GitHub Action, either in parallel or with a looping strategy, you must set the `XDG_CACHE_HOME` environment variable.
+
+The default value of this variable is `/home/ubuntu/.cache`; however, the `XDG_CACHE_HOME` variable must have a different value for each instance of the Action. If you have separate steps running in parallel, you can assign distinct values to each step, such as `XDG_CACHE_HOME: /home/ubuntu/.cache1`. If you apply a looping strategy to repeat one step multiple times, you can use an expressions to generate distinct values, such as `XDG_CACHE_HOME: /home/ubuntu/.cache<+step.identifier>`.
```
Comment (brian-f-harness) [inline docs/continuous-integration/use-ci/use-drone-plugins/run-a-git-hub-action-in-cie.md:148]:
```suggestion
The default value of this variable is `/home/ubuntu/.cache`; however, the `XDG_CACHE_HOME` variable must have a different value for each instance of the Action. If you have separate steps running in parallel, you can assign distinct values to each step, such as `XDG_CACHE_HOME: /home/ubuntu/.cache1`. If you apply a looping strategy to repeat one step multiple times, you can use an expression to generate distinct values, such as `XDG_CACHE_HOME: /home/ubuntu/.cache<+step.identifier>`.
```

**Review by brian-f-harness — DISMISSED** [review]
LGTM. One typo and a couple minor suggestions.

**Pull request merged**